File: employee_scheduler.py

import random

DAYS = ["Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday", "Sunday"]
SHIFTS = ["Morning", "Afternoon", "Evening"]
MIN_EMPLOYEES_PER_SHIFT = 2
MAX_DAYS_PER_WEEK = 5

class Scheduler:
    def __init__(self, employees, seed=42):
        self.employees = employees
        self.random = random.Random(seed)
        self.schedule = {day: {shift: [] for shift in SHIFTS} for day in DAYS}
        self.assigned_day = {name: {day: False for day in DAYS} for name in employees}
        self.work_days = {name: 0 for name in employees}
        self.logs = []

    def can_work(self, employee, day):
        return (not self.assigned_day[employee][day]) and self.work_days[employee] < MAX_DAYS_PER_WEEK

    def assign(self, employee, day, shift, reason=None):
        if self.can_work(employee, day):
            self.schedule[day][shift].append(employee)
            self.assigned_day[employee][day] = True
            self.work_days[employee] += 1
            if reason:
                self.logs.append(reason)
            return True
        return False

    def available_employees(self, day):
        return [employee for employee in self.employees if self.can_work(employee, day)]

    def assign_shift_for_day(self, day):
        unassigned = self.available_employees(day)

        for shift in SHIFTS:
            while len(self.schedule[day][shift]) < MIN_EMPLOYEES_PER_SHIFT:
                candidates = [
                    employee for employee in unassigned
                    if self.employees[employee].get(day, SHIFTS)[0] == shift
                ]

                if not candidates:
                    candidates = [
                        employee for employee in unassigned
                        if shift in self.employees[employee].get(day, SHIFTS)
                    ]

                if not candidates:
                    break

                candidates.sort(key=lambda name: (self.work_days[name], self.employees[name][day].index(shift), name))
                chosen = candidates[0]
                pref_index = self.employees[chosen][day].index(shift)

                if pref_index == 0:
                    reason = None
                else:
                    reason = (
                        f"Conflict resolved: {chosen} preferred {self.employees[chosen][day][0]} on {day}, "
                        f"but was moved to {shift} (preference rank {pref_index + 1})"
                    )

                self.assign(chosen, day, shift, reason)
                unassigned.remove(chosen)

        # If any shifts are still below the minimum, fill randomly from remaining employees.
        for shift in SHIFTS:
            while len(self.schedule[day][shift]) < MIN_EMPLOYEES_PER_SHIFT:
                remaining = self.available_employees(day)
                if not remaining:
                    break

                remaining.sort(key=lambda name: (self.work_days[name], name))
                lowest_count = self.work_days[remaining[0]]
                pool = [name for name in remaining if self.work_days[name] == lowest_count]
                chosen = self.random.choice(pool)
                self.assign(
                    chosen,
                    day,
                    shift,
                    f"Random fill: {chosen} added to {day} {shift} to satisfy minimum staffing"
                )

        # Optional extra placement for employees who could not be used on this day:
        # try the next day if one exists and a shift there is still understaffed.
        for employee in list(unassigned):
            if self.work_days[employee] >= MAX_DAYS_PER_WEEK:
                continue
            day_index = DAYS.index(day)
            if day_index + 1 >= len(DAYS):
                continue

            next_day = DAYS[day_index + 1]
            if self.assigned_day[employee][next_day]:
                continue

            for next_shift in self.employees[employee].get(next_day, SHIFTS):
                if len(self.schedule[next_day][next_shift]) < MIN_EMPLOYEES_PER_SHIFT:
                    self.assign(
                        employee,
                        next_day,
                        next_shift,
                        f"Next-day move: {employee} could not be placed on {day}, so they were assigned to {next_day} {next_shift}"
                    )
                    break

    def build(self):
        for day in DAYS:
            self.assign_shift_for_day(day)
        return self.schedule

    def print_schedule(self):
        print("=" * 72)
        print("FINAL EMPLOYEE SCHEDULE")
        print("=" * 72)
        for day in DAYS:
            print(f"\n{day}")
            print("-" * 72)
            for shift in SHIFTS:
                workers = ", ".join(self.schedule[day][shift]) if self.schedule[day][shift] else "No assignment"
                print(f"{shift:<10}: {workers}")

        print("\n" + "=" * 72)
        print("EMPLOYEE DAYS WORKED")
        print("=" * 72)
        for employee, days_worked in sorted(self.work_days.items()):
            print(f"{employee:<10} -> {days_worked} day(s)")

        print("\n" + "=" * 72)
        print("CONFLICT / AUTO-ASSIGNMENT LOG")
        print("=" * 72)
        if self.logs:
            for item in self.logs:
                print(f"- {item}")
        else:
            print("No conflicts detected.")

def get_sample_employees():
    return {
        "Alice": {
            "Monday": ["Morning", "Afternoon", "Evening"],
            "Tuesday": ["Morning", "Evening", "Afternoon"],
            "Wednesday": ["Afternoon", "Morning", "Evening"],
            "Thursday": ["Morning", "Afternoon", "Evening"],
            "Friday": ["Morning", "Evening", "Afternoon"],
            "Saturday": ["Evening", "Afternoon", "Morning"],
            "Sunday": ["Morning", "Afternoon", "Evening"],
        },
        "Bob": {
            "Monday": ["Morning", "Evening", "Afternoon"],
            "Tuesday": ["Afternoon", "Morning", "Evening"],
            "Wednesday": ["Morning", "Afternoon", "Evening"],
            "Thursday": ["Evening", "Afternoon", "Morning"],
            "Friday": ["Afternoon", "Morning", "Evening"],
            "Saturday": ["Morning", "Afternoon", "Evening"],
            "Sunday": ["Evening", "Morning", "Afternoon"],
        },
        "Carol": {
            "Monday": ["Afternoon", "Morning", "Evening"],
            "Tuesday": ["Afternoon", "Evening", "Morning"],
            "Wednesday": ["Evening", "Afternoon", "Morning"],
            "Thursday": ["Afternoon", "Morning", "Evening"],
            "Friday": ["Evening", "Afternoon", "Morning"],
            "Saturday": ["Afternoon", "Morning", "Evening"],
            "Sunday": ["Morning", "Evening", "Afternoon"],
        },
        "David": {
            "Monday": ["Evening", "Afternoon", "Morning"],
            "Tuesday": ["Morning", "Afternoon", "Evening"],
            "Wednesday": ["Evening", "Morning", "Afternoon"],
            "Thursday": ["Morning", "Evening", "Afternoon"],
            "Friday": ["Morning", "Afternoon", "Evening"],
            "Saturday": ["Evening", "Morning", "Afternoon"],
            "Sunday": ["Afternoon", "Morning", "Evening"],
        },
        "Emma": {
            "Monday": ["Morning", "Afternoon", "Evening"],
            "Tuesday": ["Evening", "Morning", "Afternoon"],
            "Wednesday": ["Morning", "Afternoon", "Evening"],
            "Thursday": ["Afternoon", "Evening", "Morning"],
            "Friday": ["Morning", "Afternoon", "Evening"],
            "Saturday": ["Afternoon", "Evening", "Morning"],
            "Sunday": ["Evening", "Afternoon", "Morning"],
        },
        "Frank": {
            "Monday": ["Afternoon", "Evening", "Morning"],
            "Tuesday": ["Morning", "Afternoon", "Evening"],
            "Wednesday": ["Afternoon", "Morning", "Evening"],
            "Thursday": ["Evening", "Morning", "Afternoon"],
            "Friday": ["Afternoon", "Evening", "Morning"],
            "Saturday": ["Morning", "Evening", "Afternoon"],
            "Sunday": ["Morning", "Afternoon", "Evening"],
        },
        "Grace": {
            "Monday": ["Evening", "Morning", "Afternoon"],
            "Tuesday": ["Afternoon", "Morning", "Evening"],
            "Wednesday": ["Morning", "Evening", "Afternoon"],
            "Thursday": ["Afternoon", "Morning", "Evening"],
            "Friday": ["Evening", "Morning", "Afternoon"],
            "Saturday": ["Morning", "Afternoon", "Evening"],
            "Sunday": ["Afternoon", "Evening", "Morning"],
        },
        "Henry": {
            "Monday": ["Morning", "Afternoon", "Evening"],
            "Tuesday": ["Morning", "Afternoon", "Evening"],
            "Wednesday": ["Evening", "Afternoon", "Morning"],
            "Thursday": ["Morning", "Afternoon", "Evening"],
            "Friday": ["Afternoon", "Morning", "Evening"],
            "Saturday": ["Evening", "Afternoon", "Morning"],
            "Sunday": ["Morning", "Evening", "Afternoon"],
        },
        "Ivy": {
            "Monday": ["Afternoon", "Morning", "Evening"],
            "Tuesday": ["Evening", "Afternoon", "Morning"],
            "Wednesday": ["Afternoon", "Morning", "Evening"],
            "Thursday": ["Morning", "Evening", "Afternoon"],
            "Friday": ["Evening", "Morning", "Afternoon"],
            "Saturday": ["Afternoon", "Morning", "Evening"],
            "Sunday": ["Evening", "Morning", "Afternoon"],
        },
        "Jack": {
            "Monday": ["Evening", "Afternoon", "Morning"],
            "Tuesday": ["Morning", "Evening", "Afternoon"],
            "Wednesday": ["Morning", "Afternoon", "Evening"],
            "Thursday": ["Evening", "Morning", "Afternoon"],
            "Friday": ["Morning", "Afternoon", "Evening"],
            "Saturday": ["Afternoon", "Evening", "Morning"],
            "Sunday": ["Morning", "Afternoon", "Evening"],
        },
    }

def get_manual_input():
    employees = {}
    number_of_employees = int(input("Enter number of employees: "))

    for _ in range(number_of_employees):
        name = input("Employee name: ").strip()
        employees[name] = {}
        print(f"Enter ranked shift preferences for {name}.")
        print("Use one line per day with three comma-separated values, for example: Morning,Evening,Afternoon")

        for day in DAYS:
            raw = input(f"{day}: ").strip()
            prefs = [part.strip().title() for part in raw.split(",") if part.strip()]
            validated = [shift for shift in prefs if shift in SHIFTS]

            for shift in SHIFTS:
                if shift not in validated:
                    validated.append(shift)

            employees[name][day] = validated[:3]

    return employees

def main():
    print("Employee Schedule Manager")
    print("1. Use sample data")
    print("2. Enter employees manually")
    choice = input("Choose an option (1 or 2): ").strip()

    employees = get_manual_input() if choice == "2" else get_sample_employees()
    scheduler = Scheduler(employees, seed=42)
    scheduler.build()
    scheduler.print_schedule()

if __name__ == "__main__":
    main()

File: EmployeeScheduler.java

import java.util.*;

public class EmployeeScheduler {
    private static final String[] DAYS = {
        "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday", "Sunday"
    };
    private static final String[] SHIFTS = {"Morning", "Afternoon", "Evening"};
    private static final int MIN_EMPLOYEES_PER_SHIFT = 2;
    private static final int MAX_DAYS_PER_WEEK = 5;

    private final Map<String, Map<String, List<String>>> employees;
    private final Map<String, Map<String, List<String>>> schedule;
    private final Map<String, Map<String, Boolean>> assignedDay;
    private final Map<String, Integer> workDays;
    private final List<String> logs;
    private final Random random;

    public EmployeeScheduler(Map<String, Map<String, List<String>>> employees, long seed) {
        this.employees = employees;
        this.schedule = new LinkedHashMap<>();
        this.assignedDay = new LinkedHashMap<>();
        this.workDays = new LinkedHashMap<>();
        this.logs = new ArrayList<>();
        this.random = new Random(seed);

        for (String day : DAYS) {
            Map<String, List<String>> shiftMap = new LinkedHashMap<>();
            for (String shift : SHIFTS) {
                shiftMap.put(shift, new ArrayList<>());
            }
            schedule.put(day, shiftMap);
        }

        for (String employee : employees.keySet()) {
            Map<String, Boolean> dayMap = new LinkedHashMap<>();
            for (String day : DAYS) {
                dayMap.put(day, false);
            }
            assignedDay.put(employee, dayMap);
            workDays.put(employee, 0);
        }
    }

    private boolean canWork(String employee, String day) {
        return !assignedDay.get(employee).get(day) && workDays.get(employee) < MAX_DAYS_PER_WEEK;
    }

    private boolean assign(String employee, String day, String shift, String reason) {
        if (canWork(employee, day)) {
            schedule.get(day).get(shift).add(employee);
            assignedDay.get(employee).put(day, true);
            workDays.put(employee, workDays.get(employee) + 1);
            if (reason != null) {
                logs.add(reason);
            }
            return true;
        }
        return false;
    }

    private List<String> availableEmployees(String day) {
        List<String> result = new ArrayList<>();
        for (String employee : employees.keySet()) {
            if (canWork(employee, day)) {
                result.add(employee);
            }
        }
        return result;
    }

    private void assignShiftForDay(String day) {
        List<String> unassigned = availableEmployees(day);

        for (String shift : SHIFTS) {
            while (schedule.get(day).get(shift).size() < MIN_EMPLOYEES_PER_SHIFT) {
                List<String> candidates = new ArrayList<>();
                for (String employee : unassigned) {
                    if (employees.get(employee).getOrDefault(day, Arrays.asList(SHIFTS)).get(0).equals(shift)) {
                        candidates.add(employee);
                    }
                }

                if (candidates.isEmpty()) {
                    for (String employee : unassigned) {
                        if (employees.get(employee).getOrDefault(day, Arrays.asList(SHIFTS)).contains(shift)) {
                            candidates.add(employee);
                        }
                    }
                }

                if (candidates.isEmpty()) {
                    break;
                }

                candidates.sort((a, b) -> {
                    int workCompare = Integer.compare(workDays.get(a), workDays.get(b));
                    if (workCompare != 0) return workCompare;

                    int prefCompare = Integer.compare(
                        employees.get(a).get(day).indexOf(shift),
                        employees.get(b).get(day).indexOf(shift)
                    );
                    if (prefCompare != 0) return prefCompare;

                    return a.compareTo(b);
                });

                String chosen = candidates.get(0);
                int prefIndex = employees.get(chosen).get(day).indexOf(shift);
                String reason = null;

                if (prefIndex > 0) {
                    reason = "Conflict resolved: " + chosen + " preferred "
                        + employees.get(chosen).get(day).get(0) + " on " + day
                        + ", but was moved to " + shift + " (preference rank " + (prefIndex + 1) + ")";
                }

                assign(chosen, day, shift, reason);
                unassigned.remove(chosen);
            }
        }

        for (String shift : SHIFTS) {
            while (schedule.get(day).get(shift).size() < MIN_EMPLOYEES_PER_SHIFT) {
                List<String> remaining = availableEmployees(day);
                if (remaining.isEmpty()) {
                    break;
                }

                remaining.sort((a, b) -> {
                    int workCompare = Integer.compare(workDays.get(a), workDays.get(b));
                    if (workCompare != 0) return workCompare;
                    return a.compareTo(b);
                });

                int lowestCount = workDays.get(remaining.get(0));
                List<String> pool = new ArrayList<>();
                for (String employee : remaining) {
                    if (workDays.get(employee) == lowestCount) {
                        pool.add(employee);
                    }
                }

                String chosen = pool.get(random.nextInt(pool.size()));
                assign(chosen, day, shift,
                    "Random fill: " + chosen + " added to " + day + " " + shift + " to satisfy minimum staffing");
            }
        }

        for (String employee : new ArrayList<>(unassigned)) {
            if (workDays.get(employee) >= MAX_DAYS_PER_WEEK) {
                continue;
            }

            int dayIndex = Arrays.asList(DAYS).indexOf(day);
            if (dayIndex + 1 >= DAYS.length) {
                continue;
            }

            String nextDay = DAYS[dayIndex + 1];
            if (assignedDay.get(employee).get(nextDay)) {
                continue;
            }

            for (String nextShift : employees.get(employee).getOrDefault(nextDay, Arrays.asList(SHIFTS))) {
                if (schedule.get(nextDay).get(nextShift).size() < MIN_EMPLOYEES_PER_SHIFT) {
                    boolean placed = assign(
                        employee,
                        nextDay,
                        nextShift,
                        "Next-day move: " + employee + " could not be placed on " + day
                            + ", so they were assigned to " + nextDay + " " + nextShift
                    );
                    if (placed) {
                        break;
                    }
                }
            }
        }
    }

    public void build() {
        for (String day : DAYS) {
            assignShiftForDay(day);
        }
    }

    public void printSchedule() {
        System.out.println("========================================================================");
        System.out.println("FINAL EMPLOYEE SCHEDULE");
        System.out.println("========================================================================");

        for (String day : DAYS) {
            System.out.println("\n" + day);
            System.out.println("------------------------------------------------------------------------");
            for (String shift : SHIFTS) {
                List<String> workers = schedule.get(day).get(shift);
                String workerList = workers.isEmpty() ? "No assignment" : String.join(", ", workers);
                System.out.printf("%-10s: %s%n", shift, workerList);
            }
        }

        System.out.println("\n========================================================================");
        System.out.println("EMPLOYEE DAYS WORKED");
        System.out.println("========================================================================");
        List<String> names = new ArrayList<>(workDays.keySet());
        Collections.sort(names);
        for (String name : names) {
            System.out.printf("%-10s -> %d day(s)%n", name, workDays.get(name));
        }

        System.out.println("\n========================================================================");
        System.out.println("CONFLICT / AUTO-ASSIGNMENT LOG");
        System.out.println("========================================================================");
        if (logs.isEmpty()) {
            System.out.println("No conflicts detected.");
        } else {
            for (String log : logs) {
                System.out.println("- " + log);
            }
        }
    }

    private static List<String> readPreferenceLine(Scanner scanner, String day) {
        System.out.print(day + ": ");
        String raw = scanner.nextLine().trim();
        String[] parts = raw.split(",");
        List<String> preferences = new ArrayList<>();

        for (String part : parts) {
            String cleaned = part.trim();
            if (!cleaned.isEmpty()) {
                String normalized = cleaned.substring(0, 1).toUpperCase() + cleaned.substring(1).toLowerCase();
                if (Arrays.asList(SHIFTS).contains(normalized) && !preferences.contains(normalized)) {
                    preferences.add(normalized);
                }
            }
        }

        for (String shift : SHIFTS) {
            if (!preferences.contains(shift)) {
                preferences.add(shift);
            }
        }

        return new ArrayList<>(preferences.subList(0, 3));
    }

    private static Map<String, Map<String, List<String>>> getManualInput() {
        Scanner scanner = new Scanner(System.in);
        Map<String, Map<String, List<String>>> employees = new LinkedHashMap<>();

        System.out.print("Enter number of employees: ");
        int numberOfEmployees = Integer.parseInt(scanner.nextLine().trim());

        for (int i = 0; i < numberOfEmployees; i++) {
            System.out.print("Employee name: ");
            String name = scanner.nextLine().trim();
            Map<String, List<String>> weeklyPrefs = new LinkedHashMap<>();

            System.out.println("Enter ranked shift preferences for " + name + ".");
            System.out.println("Use one line per day with three comma-separated values, for example: Morning,Evening,Afternoon");

            for (String day : DAYS) {
                weeklyPrefs.put(day, readPreferenceLine(scanner, day));
            }

            employees.put(name, weeklyPrefs);
        }

        return employees;
    }

    private static Map<String, Map<String, List<String>>> getSampleEmployees() {
        Map<String, Map<String, List<String>>> data = new LinkedHashMap<>();

        data.put("Alice", createWeek(
            arr("Morning", "Afternoon", "Evening"),
            arr("Morning", "Evening", "Afternoon"),
            arr("Afternoon", "Morning", "Evening"),
            arr("Morning", "Afternoon", "Evening"),
            arr("Morning", "Evening", "Afternoon"),
            arr("Evening", "Afternoon", "Morning"),
            arr("Morning", "Afternoon", "Evening")
        ));

        data.put("Bob", createWeek(
            arr("Morning", "Evening", "Afternoon"),
            arr("Afternoon", "Morning", "Evening"),
            arr("Morning", "Afternoon", "Evening"),
            arr("Evening", "Afternoon", "Morning"),
            arr("Afternoon", "Morning", "Evening"),
            arr("Morning", "Afternoon", "Evening"),
            arr("Evening", "Morning", "Afternoon")
        ));

        data.put("Carol", createWeek(
            arr("Afternoon", "Morning", "Evening"),
            arr("Afternoon", "Evening", "Morning"),
            arr("Evening", "Afternoon", "Morning"),
            arr("Afternoon", "Morning", "Evening"),
            arr("Evening", "Afternoon", "Morning"),
            arr("Afternoon", "Morning", "Evening"),
            arr("Morning", "Evening", "Afternoon")
        ));

        data.put("David", createWeek(
            arr("Evening", "Afternoon", "Morning"),
            arr("Morning", "Afternoon", "Evening"),
            arr("Evening", "Morning", "Afternoon"),
            arr("Morning", "Evening", "Afternoon"),
            arr("Morning", "Afternoon", "Evening"),
            arr("Evening", "Morning", "Afternoon"),
            arr("Afternoon", "Morning", "Evening")
        ));

        data.put("Emma", createWeek(
            arr("Morning", "Afternoon", "Evening"),
            arr("Evening", "Morning", "Afternoon"),
            arr("Morning", "Afternoon", "Evening"),
            arr("Afternoon", "Evening", "Morning"),
            arr("Morning", "Afternoon", "Evening"),
            arr("Afternoon", "Evening", "Morning"),
            arr("Evening", "Afternoon", "Morning")
        ));

        data.put("Frank", createWeek(
            arr("Afternoon", "Evening", "Morning"),
            arr("Morning", "Afternoon", "Evening"),
            arr("Afternoon", "Morning", "Evening"),
            arr("Evening", "Morning", "Afternoon"),
            arr("Afternoon", "Evening", "Morning"),
            arr("Morning", "Evening", "Afternoon"),
            arr("Morning", "Afternoon", "Evening")
        ));

        data.put("Grace", createWeek(
            arr("Evening", "Morning", "Afternoon"),
            arr("Afternoon", "Morning", "Evening"),
            arr("Morning", "Evening", "Afternoon"),
            arr("Afternoon", "Morning", "Evening"),
            arr("Evening", "Morning", "Afternoon"),
            arr("Morning", "Afternoon", "Evening"),
            arr("Afternoon", "Evening", "Morning")
        ));

        data.put("Henry", createWeek(
            arr("Morning", "Afternoon", "Evening"),
            arr("Morning", "Afternoon", "Evening"),
            arr("Evening", "Afternoon", "Morning"),
            arr("Morning", "Afternoon", "Evening"),
            arr("Afternoon", "Morning", "Evening"),
            arr("Evening", "Afternoon", "Morning"),
            arr("Morning", "Evening", "Afternoon")
        ));

        data.put("Ivy", createWeek(
            arr("Afternoon", "Morning", "Evening"),
            arr("Evening", "Afternoon", "Morning"),
            arr("Afternoon", "Morning", "Evening"),
            arr("Morning", "Evening", "Afternoon"),
            arr("Evening", "Morning", "Afternoon"),
            arr("Afternoon", "Morning", "Evening"),
            arr("Evening", "Morning", "Afternoon")
        ));

        data.put("Jack", createWeek(
            arr("Evening", "Afternoon", "Morning"),
            arr("Morning", "Evening", "Afternoon"),
            arr("Morning", "Afternoon", "Evening"),
            arr("Evening", "Morning", "Afternoon"),
            arr("Morning", "Afternoon", "Evening"),
            arr("Afternoon", "Evening", "Morning"),
            arr("Morning", "Afternoon", "Evening")
        ));

        return data;
    }

    private static Map<String, List<String>> createWeek(
        List<String> monday, List<String> tuesday, List<String> wednesday,
        List<String> thursday, List<String> friday, List<String> saturday, List<String> sunday
    ) {
        Map<String, List<String>> week = new LinkedHashMap<>();
        week.put("Monday", monday);
        week.put("Tuesday", tuesday);
        week.put("Wednesday", wednesday);
        week.put("Thursday", thursday);
        week.put("Friday", friday);
        week.put("Saturday", saturday);
        week.put("Sunday", sunday);
        return week;
    }

    private static List<String> arr(String a, String b, String c) {
        return new ArrayList<>(Arrays.asList(a, b, c));
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.println("Employee Schedule Manager");
        System.out.println("1. Use sample data");
        System.out.println("2. Enter employees manually");
        System.out.print("Choose an option (1 or 2): ");
        String choice = scanner.nextLine().trim();

        Map<String, Map<String, List<String>>> employees;
        if ("2".equals(choice)) {
            employees = getManualInput();
        } else {
            employees = getSampleEmployees();
        }

        EmployeeScheduler scheduler = new EmployeeScheduler(employees, 42L);
        scheduler.build();
        scheduler.printSchedule();
    }
}
